modules/accounts/calendar.nix

```diff
@@ -0,0 +1,193 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.accounts.calendar;
+
+  localModule = name: types.submodule {
+    options = {
+      path = mkOption {
+        type = types.str;
+        default = "${cfg.basePath}/${name}";
+        description = "The path of the storage.";
+      };
+
+      type = mkOption {
+        type = types.enum [ "filesystem" "singlefile" ];
+        description = "The type of the storage.";
+      };
+
+      fileExt = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The file extension to use.";
+      };
+
+      encoding = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          File encoding for items, both content and file name.
+          Defaults to UTF-8.
+        '';
+      };
+    };
+  };
+
+  remoteModule = types.submodule {
+    options = {
+      type = mkOption {
+        type = types.enum [ "caldav" "http" "google_calendar" ];
+        description = "The type of the storage.";
+      };
+
+      url = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The url of the storage.";
+      };
+
+      userName = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "User name for authentication.";
+      };
+
+      userNameCommand = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = [ "~/get-username.sh" ];
+        description = ''
+          A command that prints the user name to standard
+          output.
+        '';
+      };
+
+      password = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Password for authentication.";
+      };
+
+      passwordCommand = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = [ "pass" "caldav" ];
+        description = ''
+          A command that prints the password to standard
+          output.
+        '';
+      };
+
+      passwordPrompt = mkOption {
```

Comment:
I suspect this is pretty specific for vdirsyncer, perhaps could remove this as well. Alternatively move to the vdirsyncer-account file?